sceptre/plan/actions.py

```diff
@@ -1000,3 +1000,115 @@ def diff(self, stack_differ):
         :rtype: sceptre.diffing.stack_differ.StackDiff
         """
         return stack_differ.diff(self)
+
+    @add_stack_hooks
+    def detect_stack_drift(self):
+        """
+        Detects stack drift for a running stack.
+
+        :returns: The stack drift.
+        :rtype: Tuple[str, Union[str, dict]]
+        """
+        response = self._detect_stack_drift()
+
+        detection_id = response["StackDriftDetectionId"]
+        status = self._wait_for_drift(detection_id)
+
+        if status == "DETECTION_COMPLETE":
+            response = self._describe_stack_resource_drifts()
+            return_value = (self.stack.external_name, response)
+        else:
+            return_value = (self.stack.external_name, status)
+
+        return return_value
+
+    def _wait_for_drift(self, detection_id):
+        """
+        Waits for drift detection to complete.
+
+        :param detection_id: The drift detection ID.
+        :type detection_id: str
+
+        :returns: The drift status.
+        :rtype: str
+        """
+        timeout = 300
+        elapsed = 0
+
+        while True:
+            if elapsed >= timeout:
+                return "TIMED_OUT"
+
+            self.logger.debug("%s - Waiting for drift detection", self.stack.name)
```

Comment:
We should use f-strings whenever possible. They're more efficient in terms of string rendering, but, more importantly, they're really the direction of the language in terms of string formatting going forward.

Reply:
done.